app/mailers/application_mailer.rb

```diff
@@ -2,6 +2,14 @@ class ApplicationMailer < Mail::Notify::Mailer
   default from: 'from@example.com'
   layout 'mailer'
 
+  def template_mail(template_id, headers)
+    headers[:to] = Array(headers[:to]) - User.deleted.pluck(:email_address)
```

Comment:
Scoping the finder to just by the expected recipients would mean we wouldn't need all the deleted users for every mail:

```suggestion
    headers[:to] = Array(headers[:to]) - User.deleted.where(email_address: headers[:to]).pluck(:email_address)
```

This still works regardless of whether `headers[:to]` is a string, array, or nil.